docs/integration-tests.md

```diff
@@ -131,10 +128,60 @@ Requires Visual Studio 2022 Version 17.0 Preview 7.0 or greater.
 }
 ```
 
-## Run tests
+### Run tests
 
 1. Build the `FullAgent.sln`.
 2. Running Visual Studio as an Administrator, open the `IntegrationTests.sln` solution and build the solution. After a successful build, the tests are listed in the Visual Studio test explorer window.
 3. The recommended "Group By" order for the tests in the test explorer is `Project`, `Traits`, `Namespace`, `Class`.
 4. The main `IntegrationTests` test project is multi-targeted to both a .NET Framework and a .NET Core version to support both Windows/.NET Framework and Linux testing.  If you are running tests from Visual Studio on Windows, it is only necessary to run the .NET Framework version of the tests.  (Note: this is the runtime of the **test code**, not the test target applications.  The variant of the New Relic .NET agent (Framework/Core) being tested depends on the latter.)
 5. Run all tests or selected tests. 
+
+## Testing on Linux with dotnet test
+
+First, a few caveats:
+
+* Only tests with the `[NetCoreTest]` attribute (which sets an XUnit trait named `RuntimeFramework` to `NetCore`) can run on Linux.
```

Comment:
I believe currently only `[NetCoreTest]` tests that leverage the `ConsoleDynamicMethodFixture` are supported.

Reply:
At least on my system, I'm able to run the `HttpClientInstrumentationNetCore` test case on Linux, which uses a web test application (`AspNetCoreMvcBasicRequestsApplication`) instead of the ConsoleMF app:

```
dotnet test -f netcoreapp3.1 -c Release --filter "FullyQualifiedName~HttpClientInstrumentationNetCore&RuntimeFramework=Ne
tCore"
  Determining projects to restore...
  All projects are up-to-date for restore.
  NewRelic.Testing.Assertions -> /home/ubuntu/newrelic-dotnet-agent/tests/Agent/NewRelic.Testing.Assertions/bin/Release/netstandard2.0/NewRelic.Testing.Assertions.dll
  Shared -> /home/ubuntu/newrelic-dotnet-agent/tests/Agent/IntegrationTests/Shared/bin/Release/netstandard2.0/Shared.dll
  IntegrationTestHelpers -> /home/ubuntu/newrelic-dotnet-agent/tests/Agent/IntegrationTests/IntegrationTestHelpers/bin/Release/netstandard2.0/NewRelic.Agent.IntegrationTestHelpers.dll
  NetStandardTestLibrary -> /home/ubuntu/newrelic-dotnet-agent/tests/Agent/IntegrationTests/SharedApplications/Common/NetStandardTestLibrary/bin/Release/netstandard2.0/NetStandardTestLibrary.dll
  MultiFunctionApplicationHelpers -> /home/ubuntu/newrelic-dotnet-agent/tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/bin/Release/netcoreapp3.1/MultiFunctionApplicationHelpers.dll
  Models -> /home/ubuntu/newrelic-dotnet-agent/tests/Agent/IntegrationTests/Models/bin/Release/netstandard2.0/Models.dll
  IntegrationTests -> /home/ubuntu/newrelic-dotnet-agent/tests/Agent/IntegrationTests/IntegrationTests/bin/Release/netcoreapp3.1/NewRelic.Agent.IntegrationTests.dll
Test run for /home/ubuntu/newrelic-dotnet-agent/tests/Agent/IntegrationTests/IntegrationTests/bin/Release/netcoreapp3.1/NewRelic.Agent.IntegrationTests.dll (.NETCoreApp,Version=v3.1)
Microsoft (R) Test Execution Command Line Tool Version 17.0.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[01/21/2022 12:02:25] dotnet.exe exits with code 0
[01/21/2022 12:02:25] Successfully published /home/ubuntu/newrelic-dotnet-agent/tests/Agent/IntegrationTests/Applications/AspNetCoreMvcBasicRequestsApplication/AspNetCoreMvcBasicRequestsApplication.csproj to /tmp/IntegrationTestWorkingDirectory/HttpClientInstrumentationNetCore_fef23e9b-703a-4711-aba5-2650116f1554/AspNetCoreMvcBasicRequestsApplication
[01/21/2022 12:02:25] RemoteService.Start(): FileName=/tmp/IntegrationTestWorkingDirectory/HttpClientInstrumentationNetCore_fef23e9b-703a-4711-aba5-2650116f1554/AspNetCoreMvcBasicRequestsApplication/AspNetCoreMvcBasicRequestsApplication, Arguments=--port=53737 , WorkingDirectory=/tmp/IntegrationTestWorkingDirectory/HttpClientInstrumentationNetCore_fef23e9b-703a-4711-aba5-2650116f1554/AspNetCoreMvcBasicRequestsApplication, RedirectStandardOutput=True, RedirectStandardError=True, RedirectStandardInput=False
[01/21/2022 12:02:25] Waiting for process to start (/tmp/IntegrationTestWorkingDirectory/HttpClientInstrumentationNetCore_fef23e9b-703a-4711-aba5-2650116f1554/AspNetCoreMvcBasicRequestsApplication/AspNetCoreMvcBasicRequestsApplication.pid) ...
[01/21/2022 12:02:31] PID file /tmp/IntegrationTestWorkingDirectory/HttpClientInstrumentationNetCore_fef23e9b-703a-4711-aba5-2650116f1554/AspNetCoreMvcBasicRequestsApplication/AspNetCoreMvcBasicRequestsApplication.pid found...

Passed!  - Failed:     0, Passed:     1, Skipped:     0, Total:     1, Duration: < 1 ms - /home/ubuntu/newrelic-dotnet-agent/tests/Agent/IntegrationTests/IntegrationTests/bin/Release/netcoreapp3.1/NewRelic.Agent.IntegrationTests.dll (netcoreapp3.1)
```

Reply:
Awesome! Looks like I still had some broken things in my environment - working for me now too!